typing/typecore.ml

```diff
@@ -9362,7 +9365,8 @@ and type_immutable_array
 
 (* Typing of method call *)
 and type_send env loc explanation e met =
-  let obj = type_exp env mode_legacy e in
+  let expected_mode = expect_mode_cross_jkind Jkind.for_object mode_legacy in
```

Comment:
This is just a constant mode, right? If that's correct, maybe lift this outside of `type_send` to make the expected mode for an object in a send-expression more obviously constant.